src/main/java/net/fabricmc/loom/processors/MinecraftProcessedProvider.java

```diff
@@ -50,23 +53,27 @@ public MinecraftProcessedProvider(Project project, JarProcessorManager jarProces
 
 	@Override
 	protected void addDependencies(DependencyInfo dependency, Consumer<Runnable> postPopulationScheduler) {
-		if (jarProcessorManager.isInvalid(projectMappedJar) || isRefreshDeps()) {
+		if (jarProcessorManager.isInvalid(projectMappedCompileJar) || isRefreshDeps()) {
 			getProject().getLogger().lifecycle(":processing mapped jar");
 			invalidateJars();
 
 			try {
-				FileUtils.copyFile(super.getMappedJar(), projectMappedJar);
+				FileUtils.copyFile(super.getMappedJar(), projectMappedCompileJar);
+				FileUtils.copyFile(super.getMappedJar(), projectMappedRuntimeJar);
 			} catch (IOException e) {
 				throw new RuntimeException("Failed to copy source jar", e);
 			}
 
-			jarProcessorManager.process(projectMappedJar);
+			jarProcessorManager.process(projectMappedCompileJar);
+			jarProcessorManager.process(projectMappedRuntimeJar);
```

Comment:
No need to run the same processors on 2 jars.

I think the jar processor API should get a bit of an overhall to support this in a much nicer way.